Processor.py

```diff
@@ -80,6 +80,191 @@ def decode_instruction(self, bin_instruction):
 
             return methods._sta(self, memory_position)
 
+        elif bin_instruction == int('58', 16): # CLI
+            return methods._cli(self, None)
+
+        elif bin_instruction == int('D8', 16): # CLD
+            return methods._cld(self, None)
+
+        elif bin_instruction == int('B8', 16): # CLV
+            return methods._clv(self, None)
+
+        elif bin_instruction == int('C9', 16): # CMP Immediate
+            immediate = self.read_memo()
+            return methods._cmp(self, immediate, is_immediate=True)

```

Comment:
Não entendi o porquê desse parâmetro is_immediate=True. Se esse opcode for chamado o próximo valor não vai sempre ser um imediato

Reply:
A problemática é que o método _cmp() no caso do imediato, compara o valor do registrador A com o imediato. Porém quando você manda um endereço de memória, o método _cmp() carrega o valor da memória. Eu poderia ter feito de forma que o _cmp() sempre recebe o valor a ser comparado e o próprio decode_instruction() trata de carregar o valor caso não seja um imediato. O resultado será o mesmo de qualquer forma... Na aula conversarmos melhor.